library/src/main/java/com/novoda/downloadmanager/NotificationCreator.java

```diff
@@ -33,10 +37,22 @@ public int getId() {
 
             @Override
             public Notification getNotification() {
-                NotificationCompat.Builder builder = new NotificationCompat.Builder(applicationContext, notificationChannelProvider.channelId());
+                int newNotificationId = getId();
+                NotificationCustomizer.NotificationDisplayState newNotificationDisplayState = notificationDisplayState();
+
+                if (builder == null || hasChanged(newNotificationId, newNotificationDisplayState)) {
+                    builder = new NotificationCompat.Builder(applicationContext, notificationChannelProvider.channelId());
+                    notificationId = newNotificationId;
+                    notificationDisplayState = newNotificationDisplayState;
+                }
+
                 return notificationCustomizer.customNotificationFrom(builder, notificationPayload);
             }
 
+            private boolean hasChanged(int newNotificationId, NotificationCustomizer.NotificationDisplayState newNotificationDisplayState) {
+                return notificationId != newNotificationId && !notificationDisplayState.equals(newNotificationDisplayState);
```

Comment:
shouldn't be an `||` here?

Reply:
🤔 I'm pretty sure this only needs to be `state` too now that I think about it. If the id changes and the state doesn't then it would still be using the same customiser, the clients don't have access to the ids after-all.